File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
const solution = (s) => {
    const result = [];
    const my_dict = {};
    
    s.split('').forEach((str, idx) => {
        if (my_dict[str] === undefined) {
            result.push(-1);
        } else {
            result.push(idx - my_dict[str]);
        }
        my_dict[str] = idx;
    });
    return result;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    result = []
    for idx, str in enumerate(s):
        cur_front = 0
        for i in range(idx-1, -1, -1):
            if s[i] == str:
                cur_front += 1
                result.append(cur_front)
                break
            else:
                cur_front += 1
        else:
            result.append(-1)
    return result
        